templates/federated_reporting/nova_api.py

```diff
@@ -0,0 +1,134 @@
+#!/var/cfengine/bin/cfengine-selected-python
+import json
+import os
+import platform
+import sys
+import urllib3
+
+_WORKDIR = os.environ.get("WORKDIR", "/var/cfengine")
+_DEFAULT_SECRETS_PATH = "{}/httpd/secrets.ini".format(_WORKDIR)
+
+
+class NovaApi:
+
+    def __init__(
+        self, hostname=None, api_user="CFE_ROBOT", api_password=None, cert_path=None
+    ):
+        self.hostname = hostname or str(platform.node())
+        self.api_user = api_user
+        if api_password is None:
+          self.api_password = self._get_robot_password()
+        else:
+            self.api_password = api_password
+        if cert_path is None:
+            self.cert_path = "{}/httpd/ssl/certs/{}.cert".format(_WORKDIR, platform.node())
+        else:
+            self.cert_path = cert_path
+
+        self.http = urllib3.PoolManager(
+            cert_reqs="CERT_REQUIRED",
+            ca_certs=self.cert_path,
+        )
+        self.headers = urllib3.make_headers(
+            basic_auth="{}:{}".format(self.api_user, self.api_password)
+        )
+        self.headers["Content-Type"] = "application/json"
```

Comment:
All these `self.` attributes should start with an underscore and if they need to be accessible from outside of this class, they should have a respective `@property` defined. Otherwise somebody can create an instance of `NovaApi` and then (accidentally) change `api.hostname`, `api.api_user` or whatever which is not desired.

Reply:
ok. sounds good.